test/functional/plugin/lsp/inlay_hint_spec.lua

```diff
@@ -0,0 +1,153 @@
+      'inlay hints are applied when vim.lsp.inlay_hint.refresh() is called',
+      function()
+        exec_lua([[
+        bufnr = vim.api.nvim_get_current_buf()
+        vim.api.nvim_win_set_buf(0, bufnr)
+        client_id = vim.lsp.start({ name = 'dummy', cmd = server.cmd })
+      ]])
+
+        insert(text)
+        exec_lua([[vim.lsp.inlay_hint.refresh({bufnr = bufnr})]])
+        screen:expect({
+          grid = [[
+  auto add(int a, int b)-> int { return a + b; }              |
+                                                              |
+  int main() {                                                |
+      int x = 1;                                              |
+      int y = 2;                                              |
+      return add(a: x,b: y);                                  |
+  }                                                           |
+  ^}                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+                                                              |
+]]
+        })
+      end)
+
+    it(
+      'inlay hints are cleared when vim.lsp.inlay_hint.clear() is called',
+      function()
+        exec_lua([[
+        bufnr = vim.api.nvim_get_current_buf()
+        vim.api.nvim_win_set_buf(0, bufnr)
+        client_id = vim.lsp.start({ name = 'dummy', cmd = server.cmd })
+      ]])
+
+        insert(text)
+        exec_lua([[vim.lsp.inlay_hint.refresh({bufnr = bufnr})]])
+        screen:expect({
+          grid = [[
+  auto add(int a, int b)-> int { return a + b; }              |
+                                                              |
+  int main() {                                                |
+      int x = 1;                                              |
+      int y = 2;                                              |
+      return add(a: x,b: y);                                  |
+  }                                                           |
+  ^}                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+  ~                                                           |
+                                                              |
+]]
+        })
```

Comment:
This is not needed but I get `warning: Screen changes were received after the expected state.` if I remove this. I tried `screen:redraw_debug()` but the cleared state was the last one

@mfussenegger  @jdrouhard

Reply:
nevermind, bfredl said it's fine